reactor-pool-micrometer/src/main/java/reactor/pool/introspection/micrometer/Micrometer.java

```diff
@@ -0,0 +1,118 @@
+import reactor.pool.PoolMetricsRecorder;
+
+/**
+ * Micrometer supporting utilities for instrumentation of reactor-pool.
+ */
+public final class Micrometer {
+
+	/**
+	 * Create an {@link InstrumentedPool} that publishes metrics to a Micrometer {@link MeterRegistry} using the provided {@code metricsPrefix}
+	 * for naming meters, starting from the provided {@link PoolBuilder}.
+	 * <p>
+	 * The steps involved are as follows:
+	 * <ol>
+	 *     <li> create a {@link PoolMetricsRecorder} similar to {@link #recorder(String, MeterRegistry)} </li>
+	 *     <li> mutate the builder to use that recorder by calling {@link PoolBuilder#metricsRecorder(PoolMetricsRecorder)} </li>
+	 *     <li> create an {@link InstrumentedPool} via {@link PoolBuilder#buildPool()} </li>
+	 *     <li> instrument the {@link reactor.pool.InstrumentedPool.PoolMetrics} via {@link #gaugesOf(InstrumentedPool.PoolMetrics, String, MeterRegistry)} </li>
+	 *     <li> return that {@link InstrumentedPool} instance </li>
+	 * </ol>
+	 *
+	 * @param poolBuilder a pre-configured {@link PoolBuilder} on which to configure a {@link PoolMetricsRecorder}
+	 * @param metricsPrefix the prefix to use for naming the gauges and the recorder's meters
+	 * @param meterRegistry the registry to use for the gauges and the recorder's meters
+	 * @param <POOLABLE> the type of resources in the pool
+	 * @return a new {@link InstrumentedPool} with a Micrometer recorder and with gauges attached
+	 * @see DocumentedPoolMeters
+	 */
+	public static <POOLABLE> InstrumentedPool<POOLABLE> instrumentedPool(PoolBuilder<POOLABLE, ?> poolBuilder, String metricsPrefix, MeterRegistry meterRegistry) {
+		PoolMetricsRecorder recorder = recorder(metricsPrefix, meterRegistry);
+		InstrumentedPool<POOLABLE> pool = poolBuilder.metricsRecorder(recorder).buildPool();
+		gaugesOf(pool.metrics(), metricsPrefix, meterRegistry);
+		return pool;
+	}
+
+	/**
+	 * Register Micrometer gauges around the {@link InstrumentedPool}'s {@link reactor.pool.InstrumentedPool.PoolMetrics}, using the
+	 * provided prefix and publishing to the provided {@link MeterRegistry}.
+	 * <p>
+	 * {@link DocumentedPoolMeters} include the gauges which are:
+	 * <ul>
+	 *     <li> {@link DocumentedPoolMeters#ACQUIRED} </li>
+	 *     <li> {@link DocumentedPoolMeters#ALLOCATED}, </li>
+	 *     <li> {@link DocumentedPoolMeters#IDLE} </li>
+	 *     <li> {@link DocumentedPoolMeters#PENDING_ACQUIRE} </li>
+	 * </ul>
+	 *
+	 * @param poolMetrics the {@link reactor.pool.InstrumentedPool.PoolMetrics} to turn into gauges
+	 * @param metricsPrefix the prefix to use for naming the gauges
+	 * @param meterRegistry the registry to use for the gauges
+	 * @see DocumentedPoolMeters
+	 * @see #instrumentedPool(PoolBuilder, String, MeterRegistry)
+	 */
+	public static void gaugesOf(InstrumentedPool.PoolMetrics poolMetrics, String metricsPrefix, MeterRegistry meterRegistry) {
+		Gauge.builder(
+				DocumentedPoolMeters.ACQUIRED.getName(metricsPrefix), poolMetrics,
```

Comment:
mmh the readily available `InstrumentedPool` implementation (the one we offer) _is_ the `PoolMetrics`, so in this particular case it is rather desirable (we wouldn't want the meters to retain a reference to a whole pool which is otherwise disposed for instance).

if we keep the gauges, maybe we should call out this fact so that users can understand the behavior with any hypothetical 3rd party implementation that produces a standalone `PoolMetrics` instance

Reply:
Will this clash with the metrics in Reactor Netty, which we would like to keep? Or if we do not use this module there is no issue?